registry/remote/credentials/memory_store.go

```diff
@@ -17,23 +17,50 @@ package credentials
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"sync"
 
 	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/credentials/internal/config"
 )
 
-// memoryStore is a store that keeps credentials in memory.
-type memoryStore struct {
+// MemoryStore is a store that keeps credentials in memory.
+type MemoryStore struct {
 	store sync.Map
 }
 
 // NewMemoryStore creates a new in-memory credentials store.
 func NewMemoryStore() Store {
-	return &memoryStore{}
+	return &MemoryStore{}
+}
+
+// NewMemoryStoreFromConfig creates a new in-memory credentials store from the given configuration.
+//
+// Reference: https://docs.docker.com/engine/reference/commandline/cli/#docker-cli-configuration-file-configjson-properties
+func NewMemoryStoreFromConfig(c []byte) (*MemoryStore, error) {
+	cfg := struct {
+		Auths map[string]config.AuthConfig `json:"auths"`
+	}{}
+	if err := json.Unmarshal(c, &cfg); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal auth field: %w: %v", config.ErrInvalidConfigFormat, err)
+	}
+
+	s := &MemoryStore{}
+	for addr, auth := range cfg.Auths {
+		// Normalize the auth key to hostname.
+		hostname := config.ToHostname(addr)
+		cred, err := auth.Credential()
+		if err != nil {
+			return nil, err
+		}
+		s.store.Store(hostname, cred)
```

Comment:
@Wwwsylvia To keep the original behavior, should we keep the first item instead of the last item if we have conflict normalized hostname?

```suggestion
		_, _ = s.store.LoadOrStore(hostname, cred)
```

Reply:
LGTM